act/Makefile

```diff
@@ -3,39 +3,58 @@
 # Database SID, login credentials are provided in the environment.
 ACCOUNT=$(USERNAME)/$(PASSWORD)@$(SID)
 
+# Note: exit code of sqlplus is unreliable, so we use sql (i.e. sqlcl)
+
 act_shrine_ontology_mapping: status/table_access
 
-# Make a directory for status, i.e. completion times
-# Then as each script finishes, touch a corresponding file.
-status/README:
+
+status/table_access: status/demographics status/visit_details \
+		status/diagnosis status/labs status/procedures
+	sql $(ACCOUNT) @update_table_access.sql $(metadata_schema)
+	touch $@
+
+status/visit_details:
+	sql $(ACCOUNT) @visit_details_map.sql \
+		$(metadata_schema) $(shrine_ont_schema) $(heron_data_schema)
 	mkdir -p status
-	echo "These files represent completion status of SQL scripts." >$@
+	touch $@
 
 status/demographics: status/meta_manual_mapping
-	sqlplus $(ACCOUNT) @demographics_mapping.sql $(metadata_schema) $(shrine_ont_schema)
+	sql $(ACCOUNT) @demographics_mapping.sql $(metadata_schema) $(shrine_ont_schema)
+	mkdir -p status
 	touch $@
 
-status/meta_manual_mapping: status/README META_MANUAL_MAPPING.dat META_MANUAL_MAPPING.ctl
+status/meta_manual_mapping: META_MANUAL_MAPPING.dat META_MANUAL_MAPPING.ctl
 	sqlldr $(ACCOUNT) control=META_MANUAL_MAPPING.ctl
+	mkdir -p status
 	touch $@
 
 # link .dat as sqlldlr takes only .dat, and .csv are easier to view on github.
 META_MANUAL_MAPPING.dat: META_MANUAL_MAPPING.csv
 	ln -s $< $@
 
-status/diagnosis: status/README
-	sqlplus $(ACCOUNT) @diagnosis_mapping.sql $(metadata_schema) $(shrine_ont_schema)
+status/diagnosis:
+	sql $(ACCOUNT) @diagnosis_mapping.sql $(metadata_schema) $(shrine_ont_schema)
+	mkdir -p status
 	touch $@
 
-status/medications: status/README
-	sql $(ACCOUNT) @medication_mapping.sql $(etl_schema) $(shrine_ont_schema) ACT_MED_ALPHA_V2_121318
-	sql $(ACCOUNT) @medication_mapping.sql $(etl_schema) $(shrine_ont_schema) ACT_MED_VA_V2_092818
+status/labs:
+	sql $(ACCOUNT) @labs_mapping.sql $(metadata_schema) $(shrine_ont_schema) $(etl_schema)
+	mkdir -p status
 	touch $@
 
-status/procedures: status/README
-	sqlplus $(ACCOUNT) @procedure_ont_map.sql
+status/medications:
+	sql $(ACCOUNT) @medication_mapping.sql \
+		$(metadata_schema) $(shrine_ont_schema) $(etl_schema)  ACT_MED_ALPHA_V2_121318
+	sql $(ACCOUNT) @medication_mapping.sql \
+		$(metadata_schema) $(shrine_ont_schema) $(etl_schema) ACT_MED_VA_V2_092818
+	mkdir -p status
 	touch $@
 
-status/table_access: status/demographics status/diagnosis status/procedures
-	sqlplus $(ACCOUNT) @update_table_access.sql $(metadata_schema)
+status/procedures:
+	sql $(ACCOUNT) @procedure_ont_map.sql $(metadata_schema)
+	mkdir -p status
 	touch $@
```

Comment:
Dont' see  building COVID metadata using makefile. is that intentional ?

Reply:
no. oops.

Reply:
fixed in ba3ba63